File: 0061-rotate-list/0061-rotate-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode rotateRight(ListNode head, int k) {
        if (head == null){
            return head;
        }
        
        // find length and make circular
        
        int listLength = 0;
        ListNode dummyNode = new ListNode(-1, head);
        
        ListNode curr = dummyNode;
        
        while (curr.next != null){
            listLength += 1;
            curr = curr.next;
        }
        
        // making circular linked list
        curr.next = head;

        int distanceToHead =  listLength - (k % listLength);
        curr = dummyNode;

        for (int index = 0; index < distanceToHead; index ++){
            curr = curr.next;
        }
        
        ListNode res = curr.next;
        curr.next = null;
        return res;        
        
    }
}

File: 0061-rotate-list/0061-rotate-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def rotateRight(self, head: Optional[ListNode], k: int) -> Optional[ListNode]:
        
        if not head:
            return head

        dummyHead = ListNode(-1, head)
        curr = dummyHead
        self.length = 0

        while curr.next:
            curr = curr.next
            self.length += 1

        # make it a circular linked list
        curr.next = head
        curr = dummyHead

        for _ in range(self.length-(k % self.length)):
            curr = curr.next

        res = curr.next
        curr.next = None

        return res

